### ISSUE TITLE
Simple way of loading pre-trained models with hyperparameters and vocab frequency

### ISSUE THREAD

**Issue by ronaldtse** [secryst/secryst#20]
Currently when we save a trained model (as a checkpoint), the hyper parameters are not saved together. 

This means the user needs to have all this information in order to re-use the model:
1. pre-trained model
2. original hyperparameters
3. the input_vocab.json and target_vocab.json files

All of these should be somehow saved/loaded in/from a package so that the user just needs to take it and be able to run the model right away.

Perhaps there is some PyTorch functionality that addresses this? Or #3 ?

**Comment by ronaldtse** [comment]
Perhaps we should just have a ZIP file format that packages these files.

**Comment by ronaldtse** [comment]
Zip file format which contains data like this:
https://huggingface.co/deepset/bert-large-uncased-whole-word-masking-squad2#list-files

<img width="511" alt="Screen Shot 2020-10-12 at 10 19 15 PM" src="https://user-images.githubusercontent.com/11865/95756889-f8b4cd80-0cd8-11eb-99ba-6a562f700e8a.png">

```
my_model.zip
|
|--- metadata.yaml  
  # file that contains Secryst metadata, such as name of model, 
  # SHA hashes of contained files, SHA of package, 
  # original hyperparameters
|
|--- model.pth
  # Actual model
  # This could be a .pth file (for libtorch/pytorch) or a .onnx file (for cross language runtime)
|
|--- vocab.yaml 
  # contains input and output vocab
```
